src/sentry/integrations/slack/webhooks/event.py

```diff
@@ -159,108 +160,135 @@ def on_message(self, request: Request, slack_request: SlackDMRequest) -> Respons
-        """Returns true on success"""
-        matches: MutableMapping[LinkType, list[UnfurlableUrl]] = defaultdict(list)
+    def _get_unfurlable_links(
+        self,
+        request: Request,
+        slack_request: SlackDMRequest,
+        data: dict[str, Any],
+        organization: RpcOrganization | None,
+        logger_params: dict[str, Any],
+    ) -> dict[LinkType, list[UnfurlableUrl]]:
+        matches: dict[LinkType, list[UnfurlableUrl]] = defaultdict(list)
         links_seen = set()
 
-        data = slack_request.data.get("event", {})
-
-        logger_params = {
-            "integration_id": slack_request.integration.id,
-            "team_id": slack_request.team_id,
-            "channel_id": slack_request.channel_id,
-            "user_id": slack_request.user_id,
-            "channel": slack_request.channel_id,
-            **data,
-        }
-
-        # An unfurl may have multiple links to unfurl
         for item in data.get("links", []):
-            try:
-                url = item["url"]
-            except Exception:
-                _logger.exception("parse-link-error", extra={**logger_params, "url": url})
-                continue
-
-            link_type, args = match_link(url)
-
-            # Link can't be unfurled
-            if link_type is None or args is None:
-                continue
-
-            ois = integration_service.get_organization_integrations(
-                integration_id=slack_request.integration.id, limit=1
-            )
-            organization_id = ois[0].organization_id if len(ois) > 0 else None
-            organization_context = (
-                organization_service.get_organization_by_id(
-                    id=organization_id, user_id=None, include_projects=False, include_teams=False
-                )
-                if organization_id
-                else None
-            )
-            organization = organization_context.organization if organization_context else None
-            logger_params["organization_id"] = organization_id
-
-            if (
-                organization
-                and link_type == LinkType.DISCOVER
-                and not slack_request.has_identity
-                and features.has("organizations:discover-basic", organization, actor=request.user)
-            ):
+            with MessagingInteractionEvent(
+                interaction_type=MessagingInteractionType.PROCESS_SHARED_LINK,
+                spec=SlackMessagingSpec(),
+            ).capture() as lifecycle:
                 try:
-                    analytics.record(
-                        SlackIntegrationChartUnfurl(
-                            organization_id=organization.id,
-                            unfurls_count=0,
-                        )
+                    url = item["url"]
+                except Exception:
+                    lifecycle.record_failure("Failed to parse link", extra={**logger_params})
```

Comment:
This try/except is sort of weird. Realistically, this should only ever be a key error, and we can probably handle it a bit more gracefully with a `None` check 🤔